packages/ra-ui-materialui/src/list/FilterForm.js

```diff
@@ -180,6 +181,7 @@ const enhance = compose(
     reduxForm({
         form: 'filterForm',
         enableReinitialize: true,
+        destroyOnUnmount: false, // do not destroy to preserve state across navigation
```

Comment:
this is not a good solution

Reply:
After discussion, it is the right solution, because React 16 mounts before unmounting, so the redux-form is initialized then destroyed if we don't do that.

Reply:
Another solution would be to use a `form` identifier based on the resource, something like: 
```
withProps(({resource})=> ({
    form: `${resource}FilterForm`
})),
reduxForm({
   enableReinitialize: true,
})
```

Reply:
Yes, we thought about it. But it won’t work for the edit and create forms (which suffer from the same problem), because in that case, making the form name resource specific has nasty side effects (like breaking the dépendent input add-on).